tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,288 @@
+  const auto minus_five =
+      make_with_value<Scalar<DataVector>>(used_for_size, -5.0);
+
+  // Choose spacetime_metric randomly, but make sure the result is
+  // still invertible. To do this, start with
+  // Minkowski, and then add a 10% random perturbation.
+  auto spacetime_metric_int = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+  get<0, 0>(spacetime_metric_int) += get(minus_one);
+  get<1, 1>(spacetime_metric_int) += get(one);
+  get<2, 2>(spacetime_metric_int) += get(one);
+  get<3, 3>(spacetime_metric_int) += get(one);
+
+  auto spacetime_metric_ext = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+  get<0, 0>(spacetime_metric_ext) += get(minus_one);
+  get<1, 1>(spacetime_metric_ext) += get(one);
+  get<2, 2>(spacetime_metric_ext) += get(one);
+  get<3, 3>(spacetime_metric_ext) += get(one);
+
+  // Set pi, phi to be random (phi, pi should not need to be consistent with
+  // spacetime_metric for the flux consistency tests to pass)
+  const auto phi_int = make_with_random_values<
+      tnsr::iaa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+  const auto pi_int = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+  const auto phi_ext = make_with_random_values<
+      tnsr::iaa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+  const auto pi_ext = make_with_random_values<
+      tnsr::aa<DataVector, spatial_dim, Frame::Inertial>>(
+      nn_generator, nn_dist_pert, used_for_size);
+
+  const auto& spatial_metric_int = gr::spatial_metric(spacetime_metric_int);
+  const auto inverse_spatial_metric_int =
+      determinant_and_inverse(spatial_metric_int).second;
+  const tnsr::i<DataVector, spatial_dim, Frame::Inertial>
+      unit_normal_one_form_int = raise_or_lower_index(
+          random_unit_normal(nn_generator, spatial_metric_int),
+          spatial_metric_int);
+  tnsr::i<DataVector, spatial_dim, Frame::Inertial>
+      minus_unit_normal_one_form_int = unit_normal_one_form_int;
+  for (size_t i = 0; i < spatial_dim; ++i) {
+    minus_unit_normal_one_form_int.get(i) *= -1.0;
+  }
+
+  const auto shift_int =
+      gr::shift(spacetime_metric_int, inverse_spatial_metric_int);
+  const auto lapse_int = gr::lapse(shift_int, spacetime_metric_int);
+
+  const auto& spatial_metric_ext = gr::spatial_metric(spacetime_metric_ext);
+  const auto inverse_spatial_metric_ext =
+      determinant_and_inverse(spatial_metric_ext).second;
+  const DataVector one_over_one_form_magnitude_ext =
+      get(one) /
+      get(magnitude(unit_normal_one_form_int, inverse_spatial_metric_ext));
+  const auto unit_normal_one_form_ext = StrahlkorperGr::unit_normal_one_form(
```

Comment:
please remove this function from `StrahlkorperGr`, it is a generic function as far as I can tell; doesn't it exist elsewhere?

Reply:
divide_by() does the same thing; I'll switch to that here and in a separate PR remove StrahlkorperGr::unit_normal_one_form.

Reply:
PR #1571 removes StrahlkorperGr::unit_normal_one_form(), and I have removed it from this PR as well in my latest commit

Reply:
Please stop using `divide_by` and remove it from the code instead

Reply:
So @nilsdeppe @kidder I should leave `StrahlkorperGr::unit_normal_one_form()` and still use that in the upwind flux? I'm confused because @kidder said there is a more generic function I should use, so remove the one from StrahlkorperGr, but that @nilsdeppe said no remove divide_by instead...do you two agree on what I should do here?